apps/pwm_test/src/pwm_shell.c

```diff
@@ -0,0 +1,438 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include "os/mynewt.h"
+#include "os/os_dev.h"
+#include "console/console.h"
+#include "shell/shell.h"
+#include "pwm/pwm.h"
+#include "bsp/bsp.h"
+
+#include "pwm_shell.h"
+
+#define MAX_PWM_DEVICES (4)
+static struct pwm_dev * pwm_devs[MAX_PWM_DEVICES];
+
+extern void pwm_init(struct pwm_dev *pwm_dev, int pin);
+
+static int
+store_pwm(char *pwm_name, struct pwm_dev *dev)
+{
+    int i;
+
+    if (pwm_name[0] == 's') {
+        /* in case of spwmx*/
+        i = atoi(pwm_name + 4);
+    } else {
+        i = atoi(pwm_name + 3);
+    }
```

Comment:
you can have both `pwmx` and `spwmx` devices at the same time so this won't work well in such case

Reply:
ah you right